tests/unit/aofrw.tcl

```diff
@@ -179,26 +179,25 @@ start_server {tags {"aofrw external:skip"} overrides {aof-use-rdb-preamble no}}
     }
 
     test {BGREWRITEAOF is delayed if BGSAVE is in progress} {
-        r multi
+        r config set rdb-key-save-delay 1000000
         r bgsave
-        r bgrewriteaof
-        r info persistence
-        set res [r exec]
-        assert_match {*scheduled*} [lindex $res 1]
-        assert_match {*aof_rewrite_scheduled:1*} [lindex $res 2]
+        assert_match {*scheduled*} [r bgrewriteaof]
+        assert_match {*aof_rewrite_scheduled:1*} [r info persistence]
+        r config set rdb-key-save-delay 0
         while {[string match {*aof_rewrite_scheduled:1*} [r info persistence]]} {
             after 100
         }
     }
 
     test {BGREWRITEAOF is refused if already in progress} {
+        r config set aof-use-rdb-preamble yes
+        r config set rdb-key-save-delay 1000000
         catch {
-            r multi
             r bgrewriteaof
             r bgrewriteaof
-            r exec
         } e
         assert_match {*ERR*already*} $e
+        r config set rdb-key-save-delay 0
         while {[string match {*aof_rewrite_scheduled:1*} [r info persistence]]} {
```

Comment:
i still don't understand why we need that loop. i don't think this test ever sets the scheduled flag.

Reply:
Oh i now understand what u mean. i also think this loop is useless.

Reply:
I delete the loop. I think this loop is left over from a test on the copy by antirez at that time.

Reply:
:scream: i didn't know antirez created this test.
i guess it's very old. (not using `[s]` and `wait_for_condition` and so on.